src/server/svr_chk_owner.c

```diff
@@ -550,6 +550,11 @@ chk_rescResv_request(char *resvID, struct batch_request *preq)
 		return NULL;
 	}
 
+	if (resvID[0] == PBS_MNTNC_RESV_ID_CHAR && ! (preq->rq_perm & (ATR_DFLAG_OPWR | ATR_DFLAG_MGWR))) {
```

Comment:
I do not understand this change. Essentially svr_authorize_resvReq does the same thing right? It checks if the request is coming from an Operator or Manager, if not then it compares the reservation owner with the user requesting a rdel or ralter.

Reply:
@arungrover It is not the same. svr_authorize_resvReq() authorizes all Operators and Managers for all reservation, but this change rejects access for all regular users accessing maintenance reservation. It fixes the following problem: A user with the Manager role creates maintenance. Later, the Manager role is removed for this user but the user is still able to remove the maintenance.